docs/docs/concepts/memory.md

```diff
@@ -132,44 +155,124 @@ def summarize_conversation(state: State):
+## Update own instructions
+
+This is an example of long term memory.
+
+"Reflection" or "Meta-prompting" steps can use an LLM to generate or refine its own prompts or instructions. This approach allows the system to dynamically update and improve its own behavior, potentially leading to better performance on various tasks. This is particularly useful for tasks where the instructions are challenging to specify a priori. 
 
 Meta-prompting can use past information to update the prompt. As an example, this [Tweet generator](https://www.youtube.com/watch?v=Vn8A3BxfplE) uses meta-prompting to iteratively improve the summarization prompt used to generate high quality paper summaries for Twitter. In this case, we used a LangSmith dataset to house several papers that we wanted to summarize, generated summaries using a naive summarization prompt, manually reviewed the summaries, captured feedback from human review using the LangSmith Annotation Queue, and passed this feedback to a chat model to re-generate the summarization prompt. The process was repeated in a loop until the summaries met our criteria in human review.
 
-## Retrieving relevant information from long-term storage
+This will utilize the memory store concept above to store the updated instructions in a shared namespace. This namespace will have only a single item (unless you want to update instructions specific for each user, but that's a separate issue). This will look something like:
+
+```python
+# Node that *uses* the instructions
+def call_model(state: State, store: BaseStore):
+    instructions = store.search(("instructions",))[0]
+    # Application logic
+    prompt = prompt_template.format(instructions=instructions.value["instructions"])
+    ...
+
+
+# Node that updates instructions
+def update_instructions(state: State, store: BaseStore):
+    current_instructions = store.search(("instructions",))[0]
+    # Memory logic
+    prompt = prompt_template.format(instructions=instructions.value["instructions"], conversation=state["messages"])
+    output = llm.invoke(prompt)
+    new_instructions = output['new_instructions']
+    store.put(("instructions",), current_instructions.key, {"instructions": new_instructions})
+    ...
+```
+
+![](img/memory/update-instructions.png)
+### Remember a profile
+
+The profile is generally just a JSON blob with various key-value pairs. When remembering a profile, you will want to make sure that you are **updating** the profile each time. As a result, you will want to pass in the previous profile and ask the LLM to generate a new profile (or some JSON patch to apply to the old profile).
+
+If the profile is large, this can get tricky. You may need to split the profile into subsections and update each one individually. You also may need to fix errors if the LLM generates incorrect JSON.
+
+![](img/memory/update-profile.png)
+### Remember a list
```

Comment:
nit: list of information => multiple facts? Is there any other way of phrasing this?

It feels like we're naming the concept after a data structure rather than the concept that we use it to represent